ray-operator/controllers/ray/utils/util.go

```diff
@@ -104,29 +105,7 @@ func CheckRouteName(ctx context.Context, s string, n string) string {
 }
 
 // CheckName makes sure the name does not start with a numeric value and the total length is < 63 char
-func CheckName(s string) string {
-	maxLength := 50 // 63 - (max(8,6) + 5 ) // 6 to 8 char are consumed at the end with "-head-" or -worker- + 5 generated.
-
-	if len(s) > maxLength {
-		// shorten the name
-		offset := int(math.Abs(float64(maxLength) - float64(len(s))))
-		fmt.Printf("pod name is too long: len = %v, we will shorten it by offset = %v", len(s), offset)
-		s = s[offset:]
-	}
-
-	// cannot start with a numeric value
-	if unicode.IsDigit(rune(s[0])) {
-		s = "r" + s[1:]
-	}
-
-	// cannot start with a punctuation
-	if unicode.IsPunct(rune(s[0])) {
-		fmt.Println(s)
-		s = "r" + s[1:]
-	}
-
-	return s
-}
+var CheckName = utils.CheckName
```

Comment:
The implementation of `CheckName` has been moved to `ray-operator/apis/utils/util.go`, but we alias it back to avoid massive changes in this diff.